spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/annotation/EnableExternalTaskClient.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.client.spring.annotation;
+
+import org.camunda.bpm.client.ExternalTaskClient;
+import org.camunda.bpm.client.backoff.BackoffStrategy;
+import org.camunda.bpm.client.spring.impl.client.ClientRegistrar;
+import org.camunda.bpm.client.spring.impl.subscription.SubscriptionPostProcessor;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.annotation.AliasFor;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * <p>Annotation to bootstrap the External Task Client
+ *
+ * <p>Heads-up: <ul><li>for attributes of type {@link String}, the string <strong>$null$</strong>
+ * is reserved  and used as default value
+ * <li>for attributes of type {@link Long}, the value <strong>-1</strong> is reserved and used as
+ * default value
+ * <li>for attributes of type {@link Integer}, the value <strong>-1</strong> is reserved and used as
+ * default value
+ */
+@Documented
+@Target(TYPE)
+@Retention(RUNTIME)
+@Import({ClientRegistrar.class, SubscriptionPostProcessor.class})
+public @interface EnableExternalTaskClient {
+
+  String STRING_NULL_VALUE = "$null$";
+  long LONG_NULL_VALUE = -1;
+  int INT_NULL_VALUE = -1;
```

Comment:
Should we rather use some of those `Long#MIN_VALUE` and `Integer#MIN_VALUE` as placeholders as they are maybe less likely to collide with what users put in here? I could rather imagine someone putting in  a `-1` in the configuration rather than those other values and we would then be able to do a better validation maybe, telling them that their value of `-1` is not appropriate?
It is rather theoretical though, I admit. Feel free to reject it as well. 👍